shadowsocks/tcp.go

```diff
@@ -206,11 +213,15 @@ func (s *tcpService) Start() {
 			}()
 
 			findStartTime := time.Now()
-			keyID, clientConn, err := findAccessKey(clientConn, *s.ciphers)
+			keyID, clientConn, numBytes, err := findAccessKey(clientConn, *s.ciphers)
 			timeToCipher = time.Now().Sub(findStartTime)
 
 			if err != nil {
+				logger.Debugf("Failed to find a valid cipher after reading %v bytes: %v", numBytes, err)
+				io.Copy(ioutil.Discard, clientConn) // drain socket
 				return onet.NewConnectionError("ERR_CIPHER", "Failed to find a valid cipher", err)
+			} else {
```

Comment:
You have an early return, so you don't need an `else`.

Reply:
done